tools/codegen/gen_functionalization_type.py

```diff
@@ -444,41 +478,78 @@ def emit_registration_helper(f: NativeFunction, *, is_view: bool) -> str:
             registration_str = f'static_cast<{sig.ptr_type()}>(at::native::{native_api_name})'
         else:
             # non-composite view ops (and inplace ops) get a normal registration.
-            registration_str = f'TORCH_FN(functionalization::{wrapper_name(f.func)})'
+            if funcTarget == FunctionalizationTarget.FUNCTIONALIZATION:
+                registration_str = f'TORCH_FN(functionalization::{wrapper_name(f.func)})'
+            else:
+                assert funcTarget == FunctionalizationTarget.ADD_BACK_VIEWS
+                registration_str = f'TORCH_FN(functionalization::addbackviews::{wrapper_name(f.func)})'
         return f'm.impl("{f.func.name}", {registration_str});'
 
     # Note: for now, this logic is meant to avoid registering functionalization kernels for mobile.
     # At some point, Vulkan we'll want to use functionalization and we'll need to change this.
-    if not needs_functionalization(selector, g):
+    if not needs_functionalization(selector, g, Target.REGISTRATION, funcTarget):
         return []
 
     if isinstance(g, NativeFunctionsViewGroup):
-        view_str = [emit_registration_helper(g.view, is_view=True)]
-        if g.view_inplace is not None:
-            view_str.append(emit_registration_helper(g.view_inplace, is_view=True))
-        return view_str
+        # we always want to register composite kernels
+        if g.view.has_composite_implicit_autograd_kernel:
```

Comment:
The control flow here matches exactly the codegen logic below, is that right? This is exactly what Target was about; rather than having to keep the control flow thicket in sync, instead you just reuse the control flow and on the inside you then case on the target, so you only have to say the control flow once.